src/Data/Gdpr/GdprDataClass.ts

```diff
@@ -9,6 +10,77 @@ export const Gdpr = provideDataClass(
       return (await import('./gdprParamsFallback')).gdprParamsFallback()
     }
 
-    return { restApi }
+    // TODO: simply return { restApi }, once #11616 is resolved
+    if (!restApi.headers.Authorization) return { restApi }
+
+    return {
+      connection: {
+        async index(params) {
+          if (Object.keys(params.filters()).length > 0) {
+            throw new Error('GDPR consent does not support filtering.')
+          }
+
+          if (params.order().length > 0) {
+            throw new Error('GDPR consent does not support sorting.')
+          }
+
+          if (params.search()) {
+            throw new Error('GDPR consent does not support searching.')
+          }
+
+          const urlParams = new URLSearchParams()
+
+          const limit = params.limit()
+          if (limit) urlParams.set('_limit', limit.toString())
+
+          const continuation = params.continuation()
+          if (continuation) urlParams.set('_continuation', continuation)
+
+          const url = `${restApi.url}?${urlParams.toString()}`
+
+          const response = await fetch(url, { headers: restApi.headers })
+          if (!response.ok) {
+            const error = new DataConnectionError(
+              'Failed to fetch GDPR consent',
+            )
+            errorToast('Unable to connect to PisaSales', error)
```

Comment:
* [ ] Both messages will not help the visitor understand what happened and how to proceed.